src/python/pants/backend/native/subsystems/llvm.py

```diff
@@ -0,0 +1,14 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.binaries.binary_tool import NativeTool
+
+
+class LLVM(NativeTool):
```

Comment:
Boom! I love how succinct this can be now, even with the archive handling.

Reply:
Yeah, as someone who's not a pants guru yet I thought using BinaryUtil could be a lot simpler (not that it really _needed_ to be), and wasn't sure at all how to do it until @stuhood pointed me to your PR this week.

I was trying to turn the `archive_type` into a mixin or something, but I've hardly ever used python before contributing to pants and it was a bit difficult to think of how to do that. The fact that `BinaryTool#select()` can return either a path to an executable file or an unpacked directory of anything makes me a little queasy, but I think that can be iterated on if/when we start providing much more than just binaries with BinaryTool/Util. I'm thinking of providing headers/libraries in individual BinaryTool subclasses as a shim for the first iteration of C/C++ support, but that won't be difficult enough to require special treatment, I think, especially since that will hopefully be done in a more structured way when we Finally Solve the Problem of Native Dependencies.

Reply:
I think it's fine - it returns a path to an unpacked archive, but where the "archive" is just a single file, it returns that. I think that's perfectly intuitive.